benchmarks/CMakeLists.txt

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+set(STL_BENCHMARK_FLAVOR_DYNAMIC_OR_STATIC
+    static
+    CACHE STRING "Whether to use the dynamic or static standard library"
+)
+set(STL_BENCHMARK_FLAVOR_DEBUG_OR_RELEASE
+    release
+    CACHE STRING "Whether to use the debug or release library"
+)
+set(STL_BENCHMARK_FLAVOR_ITERATOR_DEBUGGING
+    default
+    CACHE STRING "What level of iterator debugging to use"
+)
+
+if(STL_BENCHMARK_FLAVOR_DYNAMIC_OR_STATIC MATCHES "^[Dd][Yy][Nn][Aa][Mm][Ii][Cc]$")
+    set(stl_is_dynamic ON)
+elseif(STL_BENCHMARK_FLAVOR_DYNAMIC_OR_STATIC MATCHES "^[Ss][Tt][Aa][Tt][Ii][Cc]$")
+    set(stl_is_dynamic OFF)
+else()
+    message(FATAL_ERROR
+        "Invalid setting for STL_BENCHMARK_FLAVOR_DYNAMIC_OR_STATIC (${STL_BENCHMARK_FLAVOR_DYNAMIC_OR_STATIC})\n"
+        "    expected one of 'dynamic', 'static'"
+    )
+endif()
+
+if(STL_BENCHMARK_FLAVOR_DEBUG_OR_RELEASE MATCHES "^[Rr][Ee][Ll][Ee][Aa][Ss][Ee]$")
+    set(stl_is_release ON)
+elseif(STL_BENCHMARK_FLAVOR_DEBUG_OR_RELEASE MATCHES "^[Dd][Ee][Bb][Uu][Gg]$")
+    set(stl_is_release OFF)
+else()
+    message(FATAL_ERROR
+        "Invalid setting for STL_BENCHMARK_FLAVOR_DEBUG_OR_RELEASE (${STL_BENCHMARK_FLAVOR_DEBUG_OR_RELEASE})\n"
+        "    expected one of 'release', 'debug'"
+    )
+endif()
+
+if(stl_is_release)
+    set(debug_flavor "")
+else()
+    set(debug_flavor "d")
+endif()
+
+set(iterator_debugging "")
+if(STL_BENCHMARK_FLAVOR_ITERATOR_DEBUGGING MATCHES "^[Dd][Ee][Ff][Aa][Uu][Ll][Tt]$")
+    # default
+elseif(STL_BENCHMARK_FLAVOR_ITERATOR_DEBUGGING STREQUAL "0")
+    if(NOT stl_is_release)
+        set(iterator_debugging "0")
+    endif()
+elseif(STL_BENCHMARK_FLAVOR_ITERATOR_DEBUGGING STREQUAL "1")
+    set(iterator_debugging "1")
+elseif(STL_BENCHMARK_FLAVOR_ITERATOR_DEBUGGING STREQUAL "2")
+    if(stl_is_release)
+        message(FATAL_ERROR "The release libraries do not support full iterator debugging")
+    endif()
+else()
+    message(FATAL_ERROR
+        "Invalid setting for STL_BENCHMARK_FLAVOR_ITERATOR_DEBUGGING (${STL_BENCHMARK_FLAVOR_ITERATOR_DEBUGGING})\n"
+        "    expected one of 'default', '0', '1', 2'"
+    )
+endif()
+
+if(stl_is_dynamic)
+    if(NOT iterator_debugging STREQUAL "")
+        message(FATAL_ERROR "The dynamic STL does not support non-default iterator debugging")

```

Comment:
This is not entirely accurate. `/MD` and `/MDd` support non-default IDL, except that custom facets don't work. Most of the STL's tests are able to have dynamic non-default-IDL configurations, with only a few exceptions.